web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/ImageServerProxyController.java

```diff
@@ -30,87 +28,62 @@
 public class ImageServerProxyController {
     private static final Logger LOG = LoggerFactory.getLogger(ImageServerProxyController.class);
-
     @Autowired
     private ImageServerProxyService imageServerProxyService;
-
     @Autowired
     private AccessControlService accessControlService;
 
-    /**
-     * Determines if the user is allowed to access a JP2 datastream on the selected object.
-     *
-     * @param pid
-     * @return
-     */
-    private boolean hasAccess(PID pid) {
-        var datastream = JP2_ACCESS_COPY.getId();
-
-        Permission permission = DatastreamPermissionUtil.getPermissionForDatastream(datastream);
-
-        AgentPrincipals agent = AgentPrincipalsImpl.createFromThread();
-        LOG.debug("Checking if user {} has access to {} belonging to object {}.",
-                agent.getUsername(), datastream, pid);
-        return accessControlService.hasAccess(pid, agent.getPrincipals(), permission);
-    }
-
     /**
      * Handles requests for individual region tiles.
      * @param id
      * @param region
-     * @param size
+     * @param size pixel size or max for full size
      * @param rotation
      * @param qualityFormat
-     * @param response
      */
+    @CrossOrigin
     @GetMapping("/iiif/v3/{id}/{region}/{size}/{rotation}/{qualityFormat:.+}")
-    public void getRegion(@PathVariable("id") String id,
+    public ResponseEntity<InputStreamResource> getRegion(@PathVariable("id") String id,
                           @PathVariable("region") String region,
                           @PathVariable("size") String size, @PathVariable("rotation") String rotation,
-                          @PathVariable("qualityFormat") String qualityFormat, HttpServletResponse response) {
+                          @PathVariable("qualityFormat") String qualityFormat) {
 
         PID pid = PIDs.get(id);
         // Check if the user is allowed to view this object
-        if (this.hasAccess(pid)) {
-            try {
-                String[] qualityFormatArray = qualityFormat.split("\\.");
-                String quality = qualityFormatArray[0];
-                String format = qualityFormatArray[1];
-                response.addHeader("Access-Control-Allow-Origin", "*");
-                imageServerProxyService.streamJP2(
-                        id, region, size, rotation, quality, format,
-                        response.getOutputStream(), response, 1);
-            } catch (IOException e) {
-                LOG.error("Error retrieving streaming JP2 content for {}", id, e);
-                response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
-            }
-        } else {
-            LOG.debug("Access was forbidden to {} for user {}", id, GroupsThreadStore.getUsername());
-            response.setStatus(HttpStatus.FORBIDDEN.value());
+        AccessGroupSet principals = getAgentPrincipals().getPrincipals();
+        accessControlService.assertHasAccess("Insufficient permissions to metadata for " + id,
```

Comment:
I guess this message isn't really accurate for a request to get a jpg region

Reply:
hehe whoops! fixed this and the missing word for the actual get metadata endpoint